RELEASE.md

```diff
@@ -0,0 +1,28 @@
+# Release Process
+
+The process of creating a release is described in this document. Replace `X.Y.Z` with the version to be released.
+
+## 1. Create a `releases/X.Y.Z` branch 
+
+The `releases/X.Y.Z` branch should be created from a base branch. For minor releases the base is `master` and for patch releases the base is `releases/X.Y.(Z-1)`.
+
```

Comment:
can we add a for example how to do it? `i.e., for this step use the github UI for switching branches to create a new branch.`

Reply:
Added a link to GitHub docs on that